lib/wallaroo/core/barrier/barrier_step_forwarder.pony

```diff
@@ -110,6 +123,15 @@ class BarrierStepForwarder
     check_completion(_step.inputs())
 
   fun ref check_completion(inputs: Map[RoutingId, Producer] box) =>
+    @printf[I32]("[JB] check_completion\n".cstring())
+    if (inputs.size() - _inputs_blocking.size()) == 1 then
```

Comment:
This whole block seems to have no functional purpose.  I think it's left over from a test Scott was doing and can be removed.

Reply:
Correct, it can be removed